graylog2-server/src/main/java/org/graylog2/indexer/messages/Indexable.java

```diff
@@ -38,11 +39,19 @@ public interface Indexable {
      * The message id is represented as a {@link de.huxhorn.sulky.ulid.ULID}
      */
     String getMessageId();
+
     long getSize();
+
     DateTime getReceiveTime();
-    Map<String, Object> toElasticSearchObject(ObjectMapper objectMapper,@Nonnull final Meter invalidTimestampMeter);
+
+    Map<String, Object> toElasticSearchObject(ObjectMapper objectMapper, @Nonnull final Meter invalidTimestampMeter);
+
     DateTime getTimestamp();
 
+    default byte[] serialize(ObjectMapper objectMapper, @Nonnull final Meter invalidTimestampMeter) throws JsonProcessingException {
```

Comment:
Let's pass in a "serialization config" object that contains the object mapper and the meter. That way, we can extend the serialization configuration in the future without breaking the method signature.

Reply:
Good idea. I named it `SerializationContext` because IMO it's a bit more than just config. Does that look like what you had in mind?